pint/formatting.py

```diff
@@ -287,12 +305,71 @@ def format_compact(unit: UnitsContainer, registry: UnitRegistry, **options) -> s
         division_fmt="/",
         power_fmt="{}**{}",
         parentheses_fmt=r"({})",
+        registry=registry,
         **options,
     )
 
 
+def dim_sort(items: Iterable[Tuple[str, Number]], registry: UnitRegistry):
+    """Sort a list of units by dimensional order.
+
+    Parameters
+    ----------
+    items : tuple
+        a list of tuples containing (unit names, exponent values).
+    registry : UnitRegistry
+        the registry to use for looking up the dimensions of each unit.
+
+    Returns
+    -------
+    list
+        the list of units sorted by most significant dimension first.
+
+    Raises
+    ------
+    KeyError
+        If unit cannot be found in the registry.
+    """
+    if registry is None or len(items) <= 1:
+        return items
+    # if len(items) == 2 and items[0][1] * items[1][1] < 0:
+    #     return items
+    ret_dict = dict()
+    for name, value in items:
```

Comment:
you could change this to `for unit_name, unit_exponent in items:` to make it easier to follow

Reply:
I have some other cleanups to add.  Will add that as well.